lib/game.rb

```diff
@@ -11,4 +11,22 @@ def initialize
     @cpu = Player.new
     @winner = nil
   end
+
+  def introduction
+    puts "Welcome to BATTLESHIP"
+    puts "Enter p to play. Enter q to quit."
+    input = gets.chomp
+    if input == "p"
+      set_player_name
+    elsif input == "q"
+      abort("We'll see you next time!")
```

Comment:
Awesome use of the abort method to close the program if the user is not interested in playing!